### PR TITLE
avoid_unnecessary_containers

### PR DESCRIPTION
**Pull request by pq**
See: #1560.

/cc @jacob314 @scheglov

### CODE REVIEW

File: lib/src/util/flutter_utils.dart [hunk 1]
```diff
@@ -13,9 +14,8 @@ var _collectionInterfaces = <InterfaceTypeDefinition>[
   InterfaceTypeDefinition('LinkedHashSet', 'dart.collection'),
 ];
 
-// todo (pq): consider caching lookups
-bool isWidgetType(DartType type) =>
-    DartTypeUtilities.implementsInterface(type, 'Widget', '');
+bool isContainerType(DartType type) =>
+    DartTypeUtilities.implementsInterface(type, 'Container', '');
```
Comment (scheglov) [inline lib/src/util/flutter_utils.dart]: Is this the semantic you want - not just exactly `Container`, but also its subclasses?
Why not `isInterface`?

Comment (scheglov) [reply lib/src/util/flutter_utils.dart]: See for example `isExactWidgetTypeContainer` is DAS.

Comment (pq) [reply lib/src/util/flutter_utils.dart]: hmm, yes.  i think you're right.  thanks.

Comment (pq) [reply lib/src/util/flutter_utils.dart]: Ok, fixed.  PTAL?

**Pull request merged**